hooli-demo-assets/hooli_demo_assets/resources/__init__.py

```diff
@@ -1,66 +1,99 @@
 parent_dir_path = current_file_path.parent.parent.parent.parent
 DUCKDB_PATH = parent_dir_path / "dbt_project" / "example.duckdb"
 
-source = SlingSourceConnection(
-        type="s3",
-        bucket=EnvVar("AWS_S3_BUCKET"),
-        region=EnvVar("AWS_REGION"),
-        access_key_id=EnvVar("AWS_ACCESS_KEY_ID"),
-        secret_access_key=EnvVar("AWS_SECRET_ACCESS_KEY"),
-)
-local_target = SlingTargetConnection(
-        type="duckdb",
-        instance=f"{DUCKDB_PATH}",
-)
-branch_target = SlingTargetConnection(
-        type="snowflake",
-        host=EnvVar("SNOWFLAKE_HOST"),
-        database="DEMO_DB2_BRANCH",
-        user=EnvVar("SNOWFLAKE_USER"),
-        password=EnvVar("SNOWFLAKE_PASSWORD"),
-        role=EnvVar("SNOWFLAKE_ROLE"),
-)
-prod_target = SlingTargetConnection(
-        type="snowflake",
-        host=EnvVar("SNOWFLAKE_HOST"),
-        database="DEMO_DB2",
-        user=EnvVar("SNOWFLAKE_USER"),
-        password=EnvVar("SNOWFLAKE_PASSWORD"),
-        role=EnvVar("SNOWFLAKE_ROLE"),
-)
 
-resource_def = {
-    "LOCAL": {
-         "s3_to_snowflake_resource":
-         SlingResource(source_connection=source, target_connection=local_target),
-    },
-    "BRANCH": {
-         "s3_to_snowflake_resource": SlingResource(
-            source_connection=source, target_connection=branch_target
-        ),
-    },
-    "PROD": {
-        "s3_to_snowflake_resource": SlingResource(
-            source_connection=source, target_connection=prod_target
-        ),
-    },
-}
\ No newline at end of file
+# Alternatively a replication.yaml file can be used
+def create_replication_config(env: str):
+   # Determine the target database dynamically based on the environment
+   target_database = {
+       'LOCAL': 'DUCKDB',
+       'BRANCH': 'SNOWFLAKE_BRANCH',
+       'PROD': 'SNOWFLAKE_PROD'
+   }.get(env, 'DUCKDB')  # Default to DUCKDB if no environment matches
+
+   replication_config = {
+       "source": "S3",
+       "target": target_database,
+       "defaults": {
+           "mode": "full-refresh",
+           "object": "{stream_file_folder}.{stream_file_name}",
+           "source_options": {
+               "format": "csv"
+           }
+       },
+       "streams": {
+           "s3://hooli-demo/embedded-elt/locations.csv": {
+               "object": "locations"
+           }
+       }
+   }
+   return replication_config
+
+
+def create_sling_resource(env: str):
+   # Dynamically generate connection based on enviornment
+   connections = [
+       SlingConnectionResource(
+           name="S3",
+           type="s3",
+           bucket=EnvVar("AWS_S3_BUCKET"),
+           region=EnvVar("AWS_REGION"),
+           access_key_id=EnvVar("AWS_ACCESS_KEY_ID"),
+           secret_access_key=EnvVar("AWS_SECRET_ACCESS_KEY"),
+       )
+   ]
+   if env == 'LOCAL':
+       connections.append(SlingConnectionResource(
+           name="DUCKDB",
+           type="duckdb",
+           instance=f"{DUCKDB_PATH}",
+           schema="RAW_DATA",
```

Comment:
@izzye84  This kept bothering me. I tested this locally after adding the `database` param to match the other configs, and 🎉  duckdb can materialize the asset
```suggestion
           instance=f"{DUCKDB_PATH}",
           database="example",
           schema="RAW_DATA",
```